holmes/plugins/toolsets/datadog/toolset_datadog_logs.py

```diff
@@ -0,0 +1,351 @@
+                    status=ToolResultStatus.SUCCESS,
+                    data=logs_str,
+                    params=params.model_dump(),
+                )
+            else:
+                return StructuredToolResult(
+                    status=ToolResultStatus.NO_DATA,
+                    params=params.model_dump(),
+                )
+
+        except DataDogRequestError as e:
+            logging.exception(e, exc_info=True)
+            return StructuredToolResult(
+                status=ToolResultStatus.ERROR,
+                error=f"Exception while querying Datadog: {str(e)}",
+                params=params.model_dump(),
+                invocation=json.dumps(e.payload),
+            )
+
+        except Exception as e:
+            logging.exception(
+                f"Failed to query Datadog logs for params: {params}", exc_info=True
+            )
+
+            return StructuredToolResult(
+                status=ToolResultStatus.ERROR,
+                error=f"Exception while querying Datadog: {str(e)}",
+                params=params.model_dump(),
+            )
+
+    def _perform_healthcheck(self) -> Tuple[bool, str]:
+        """
+        Perform a healthcheck by fetching a single log from Datadog.
+        Returns (success, error_message).
+        """
+        try:
+            logging.info("Performing Datadog configuration healthcheck...")
+            healthcheck_params = FetchPodLogsParams(
+                namespace="*",
+                pod_name="*",
+                limit=1,
+                start_time="-172800",  # 48 hours in seconds
+            )
+
+            result = self.fetch_pod_logs(healthcheck_params)
+
+            if result.status == ToolResultStatus.ERROR:
+                error_msg = result.error or "Unknown error during healthcheck"
+                logging.error(f"Datadog healthcheck failed: {error_msg}")
+                return False, f"Datadog healthcheck failed: {error_msg}"
+            elif result.status == ToolResultStatus.NO_DATA:
+                error_msg = "No logs were found in the last 48 hours using wildcards for pod and namespace. Is the configuration correct?"
+                logging.error(f"Datadog healthcheck failed: {error_msg}")
+                return False, f"Datadog healthcheck failed: {error_msg}"
+
+            logging.info("Datadog healthcheck completed successfully")
+            return True, ""
+
+        except Exception as e:
+            logging.exception("Failed during Datadog healthcheck")
+            return False, f"Healthcheck failed with exception: {str(e)}"
+
+    def prerequisites_callable(self, config: dict[str, Any]) -> Tuple[bool, str]:
+        if not config:
+            return (
+                False,
+                "Datadog toolset is misconfigured. 'dd_api_key' and 'dd_app_key' are required.",
+            )
+
+        try:
+            dd_config = DatadogConfig(**config)
+            if not dd_config.storage_tiers:
```

Comment:
this could be handled already by pydantic with min_items check.

https://docs.pydantic.dev/1.10/usage/types/#arguments-to-conlist